java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -124,23 +127,34 @@ private static void sanitize(UnitTestIndex index) {
 
   private void save(UnitTestIndex index, SensorContext context) {
     long negativeTimeTestNumber = 0;
-    for (Map.Entry<String, UnitTestClassReport> entry : index.getIndexByClassname().entrySet()) {
+    Map<InputFile, UnitTestClassReport> indexByInputFile = consolidateResults(index.getIndexByClassname());
+    for (Map.Entry<InputFile, UnitTestClassReport> entry : indexByInputFile.entrySet()) {
       UnitTestClassReport report = entry.getValue();
       if (report.getTests() > 0) {
         negativeTimeTestNumber += report.getNegativeTimeTestNumber();
-        InputFile resource = getUnitTestResource(entry.getKey());
-        if (resource != null) {
-          save(report, resource, context);
-        } else {
-          LOGGER.debug("Resource not found: {}", entry.getKey());
-        }
+        save(report, entry.getKey(), context);
       }
     }
     if (negativeTimeTestNumber > 0) {
       LOGGER.warn("There is {} test(s) reported with negative time by surefire, total duration may not be accurate.", negativeTimeTestNumber);
     }
   }
 
+  private Map<InputFile, UnitTestClassReport> consolidateResults(Map<String, UnitTestClassReport> indexByClassname) {
```

Comment:
I tend to think the name is a bit too generic : mapToInputFile would be more appropriate.